pyproject.toml

```
@@ -122,7 +122,7 @@ test = [
     "coverage",
     "fakeredis[lua]",
     "kaleido",
-    "moto",
+    "moto>=5.0.0",
```

I'm not familiar with this test and library, but I think there are two ways to resolve this issue for the current CI error.

The first one is to add a version restriction:

```
    "moto<5.0.0",
```

and adding some TOOD comment to remove it after dropping python 3.7 by Optuna.

Alternatively, add a version constraint more specifically: 
```suggestion
    "moto>=5.0.0; python_version>='3.8'",
```
and adding python version dependent import for `tests/artifacts_tests/test_boto3.py`.

---

Thank you for the heads-up! I was tricked because I tried to run tests on Python 3.7 and got a bunch of other errors.
Thanks for the suggestion of using the environment marker. It looked very promising, but I double-checked [PEP-508](https://peps.python.org/pep-0508) and realized that the false environment condition ignores the dependency specification.

I think there is an even more straightforward way that won't require specifying a `moto` version at all. Let me re-evaluate tests and come back with a solution :)